features/TESTS/filesystem/mbr_block_device/main.cpp

```diff
@@ -0,0 +1,226 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+
+#include "HeapBlockDevice.h"
+#include "MBRBlockDevice.h"
+#include <stdlib.h>
+
+using namespace utest::v1;
+
+#define BLOCK_COUNT 16
+#define BLOCK_SIZE 512
+
+HeapBlockDevice bd(BLOCK_COUNT*BLOCK_SIZE, BLOCK_SIZE);
+
+// Testing formatting of master boot record
+void test_mbr_format()
+{
+    // Create two partitions splitting device in ~half
+    int err = MBRBlockDevice::partition(&bd, 1, 0, 0, (BLOCK_COUNT/2)*BLOCK_SIZE);
+    TEST_ASSERT_EQUAL(0, err);
+
+    err = MBRBlockDevice::partition(&bd, 2, 0, -(BLOCK_COUNT/2)*BLOCK_SIZE);
+    TEST_ASSERT_EQUAL(0, err);
+
+    // Load both partitions, as well as a third to check for invalid partitions
+    MBRBlockDevice part1(&bd, 1);
+    err = part1.init();
+    TEST_ASSERT_EQUAL(0, err);
+
+    MBRBlockDevice part2(&bd, 2);
+    err = part2.init();
+    TEST_ASSERT_EQUAL(0, err);
+
+    MBRBlockDevice part3(&bd, 3);
+    err = part3.init();
+    TEST_ASSERT_EQUAL(BD_ERROR_INVALID_PARTITION, err);
+
+    // Deinit partitions
+    err = part1.deinit();
+    TEST_ASSERT_EQUAL(0, err);
+
+    err = part2.deinit();
+    TEST_ASSERT_EQUAL(0, err);
+}
+
+// Testing mbr attributes
+void test_mbr_attr()
+{
+    // Load partitions
+    MBRBlockDevice part1(&bd, 1);
+    int err = part1.init();
+    TEST_ASSERT_EQUAL(0, err);
+
+    MBRBlockDevice part2(&bd, 2);
+    err = part2.init();
+    TEST_ASSERT_EQUAL(0, err);
+
+    // Test attributes on partitions
+    printf("partition 1 partition number: %d\n",        part1.get_partition_number());
```

Comment:
Understood, but still the test should not print anything if is all green ? This is more for future request - having logging library. Some tests already have something like ``DEBUG_PRINT`` that is a config attribute and by default disabled. Is it time now to consider having something similar to it - a proper replacement?